shipyard-controller/db/mongodb_event_repo.go

```diff
@@ -76,6 +76,24 @@ func (mdbrepo *MongoDBEventsRepo) GetEvents(project string, filter common.EventF
 	return events, nil
 }
 
+// GetEventByID gets a single event of a project, based on the id
+func (mdbrepo *MongoDBEventsRepo) GetEventByID(projectName string, filter common.EventFilter, status ...common.EventStatus) (apimodels.KeptnContextExtendedCE, error) {
+	collection, ctx, cancel, err := mdbrepo.getEventsCollection(projectName, status...)
+	var event apimodels.KeptnContextExtendedCE
+
+	if err != nil {
+		return event, err
+	}
+
+	defer cancel()
+
+	cur := collection.FindOne(ctx, getSearchOptions(filter))
```

Comment:
please check for `cur.Err()` right after calling the function and add error handling - note that `ErrNoDocuments` should be handled differently than other kinds of errors - For this one we can return `nil, ErrNoEventFound` - for all other errors `nil, fmt.Errorf("could not retrieve event: %w", err)